Bengt/specs consistency/base ledger (#94)
anoma/namada-docs: packages/specs/pages/base-ledger/consensus.mdx

# Consensus

Namada uses [CometBFT](https://github.com/cometbft/cometbft/) (nee Tendermint Go) through the [cometbft-rs](https://github.com/heliaxdev/tendermint-rs) (nee tendermint-rs) bindings in order to provide peer-to-peer transaction gossip, Byzantine fault tolerant (BFT) consensus, and state machine replication for Namada's custom state machine. CometBFT implements the Tendermint consensus algorithm, which you can read more about [here](https://arxiv.org/abs/1807.04938).

{/* Maybe we want to leave out the below section. I just felt we should give a brief summary of CometBFT */}
## The benefits of using CometBFT

Using the CometBFT consensus algorithm comes with a number of benefits including but not limited to:

- Fast finality 

>     - Tendermint achieves fast and deterministic finality, meaning that once a block is committed to the blockchain, it is irreversible. This is crucial for applications rely on settled transactions that cannot be rolled back.

Reviewer: ```suggestion
    - CometBFT achieves fast and deterministic finality, meaning that once a block is committed to the blockchain, it is irreversible. This is crucial for applications rely on settled transactions that cannot be rolled back.
```

- Inter-blockchain communication system (IBC)

>     - Composability with all other Tendermint based blockchains, such as Cosmos-ecosystem blockchains

Reviewer: ```suggestion
    - Composability with all other CometBFT-based blockchains, such as Cosmos-ecosystem blockchains
```

- Battle tested

>     - The entire cosmos-ecosystem have been using the Tendermint 

Reviewer: ```suggestion
    - The entire Cosmos ecosystem has been using CometBFT (nee Tendermint) for years
```

- Customisable

>     - Allows the setting of various of parameters, including the ability to implement a custom proof of stake algorithm

Reviewer: ```suggestion
    - Allows the setting of various parameters, including the ability to implement a custom proof of stake algorithm
```
